lib/Onyx.js

```diff
@@ -780,6 +780,36 @@ function mergeCollection(collectionKey, collection) {
         });
 }
 
+/**
+ * Insert API responses and lifecycle data into Onyx
+ *
+ * @param {Array} data An array of objects with shape {onyxMethod: oneOf('set', 'merge'), key: string, value: *}
+ */
+function update(data) {
+    // First, validate the Onyx object is in the format we expect
+    _.each(data, ({onyxMethod, key}) => {
+        if (!_.contains(['set', 'merge'], onyxMethod)) {
+            throw new Error(`Invalid onyxMethod ${onyxMethod} in Onyx update.`);
+        }
+        if (!_.isString(key)) {
+            throw new Error(`Invalid ${typeof key} key provided in Onyx update. Onyx key must be of type string.`);
+        }
+    });
+
+    _.each(data, ({onyxMethod, key, value}) => {
+        switch (onyxMethod) {
+            case 'set':
+                set(key, value);
+                break;
+            case 'merge':
+                merge(key, value);
+                break;
+            default:
+                break;
```

Comment:
Usually, I prefer switch/case, but this seems more verbose than two simple conditional checks. NAB though

Reply:
Haha I initially had it as a series of conditionals, but changed it because of @tgolen's comments [here](https://github.com/Expensify/react-native-onyx/pull/133#discussion_r879765933). I don't have strong feeling either way.

Reply:
Sounds good! Let's not switch this then 🤣